### PR TITLE
Changing IValueProvider.GetValue() to GetValueAsync()

### PR DESCRIPTION
**Pull request by brettsam**

### CODE REVIEW

File: src/Microsoft.Azure.WebJobs.Host/Bindings/IValueProvider.cs [hunk 1]
```diff
@@ -19,7 +20,7 @@ public interface IValueProvider
         /// Gets the value.
```
Comment (mathewc) [inline src/Microsoft.Azure.WebJobs.Host/Bindings/IValueProvider.cs]: nit: Can you update the help text for these similar to what we use in other places for task returning values, e.g. IBinding.BindAsync, etc. E.g. "A task that returns the value".

File: test/Microsoft.Azure.WebJobs.Host.FunctionalTests/TableTests.cs [hunk 2]
```diff
@@ -143,7 +143,7 @@ public void Table_IfBoundToICollectorJObject_AddInsertsEntity()
             Assert.NotNull(entity.Properties);
 
             AssertPropertyValue(entity, "ValueStr", "abcdef");
-            AssertPropertyValue(entity, "ValueNum", 123);
```
Comment (mathewc) [inline test/Microsoft.Azure.WebJobs.Host.FunctionalTests/TableTests.cs]: What are all these formatting changes that your editor makes in your review?

File: test/Microsoft.Azure.WebJobs.Host.UnitTests/Singleton/SingletonValueProviderTests.cs [hunk 3]
```diff
@@ -36,9 +37,9 @@ public void Type_ReturnsExpectedValue()
         }
 
         [Fact]
-        public void GetValue_ReturnsExpectedValue()
+        public async Task GetValue_ReturnsExpectedValue()
```
Comment (mathewc) [inline test/Microsoft.Azure.WebJobs.Host.UnitTests/Singleton/SingletonValueProviderTests.cs]: nit: GetValueAsync

File: test/Microsoft.Azure.WebJobs.Host.FunctionalTests/TableTests.cs (hunk 2 above)
Comment (brettsam) [inline test/Microsoft.Azure.WebJobs.Host.FunctionalTests/TableTests.cs]: I use https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.ProductivityPowerTools2015. There's a `Format document on save` and `Remove and Sort Usings on save` that fixes spacing issues when I save a file.

Comment (brettsam) [inline test/Microsoft.Azure.WebJobs.Host.FunctionalTests/TableTests.cs]: BTW -- if it's hard to review b/c of all the spacing changes, you can add ?w=1 to the URL to ignore whitespace changes: https://github.com/Azure/azure-webjobs-sdk/pull/970/files?w=1

**Pull request merged**